packages/jaeger-ui/src/model/ddg/types.tsx

```diff
@@ -53,3 +55,44 @@ export type TDdgModel = {
 };
 
 export type TDdgVertex = TVertex<{ service: string; operation: string }>;
+
+export type TDdgStateEntry =
+  | {
+      state: typeof fetchedState.LOADING;
+    }
+  | {
+      error: ApiError;
+      state: typeof fetchedState.ERROR;
+    }
+  | {
+      model: TDdgModel;
+      state: typeof fetchedState.DONE;
+      styleStates: Map<number, number>;
+    };
+
+export const stateKey = (service: string, operation: string = '*', start: number, end: number): string =>
```

Comment:
this could just take in a kwarg object instead of needing to specify `payload.operation == null ? null : payload.operation`
that would also play nice with some other utils that are returning objects containing these four keys (or these four keys + visEncoding)

Reply:
Kwarg object is 👌